File: python/bit/q268.py
#Bit manipulation
def missingNumber1(self, nums):
    """
    :type nums: List[int]
    :rtype: int
    """
    ans = 0
    for i in range(0,len(nums)+1):
        ans ^= i
    for i in nums:
        ans ^= i
    return ans



#Math approach
def missingNumber2(self, nums):
    """
    :type nums: List[int]
    :rtype: int
    """
    sum1 = (1+len(nums))*len(nums)/2
    sum2 = sum(nums)
    return sum1-sum2

#Interesting approach
def missingNumber3(self, nums):
    """
    :type nums: List[int]
    :rtype: int
    """
    sum = 0
    for i,val in enumerate(nums):
        sum += (val-i)
    return len(nums) - sum

File: c++/bit/q268.cpp
//Bit
int missingNumber1(vector<int>& nums) {
    int ans = 0;
    for(int i = 0; i < nums.size(); i++){
        ans ^= i;
        ans ^= nums[i];
    }

    ans ^= nums.size();
    return ans;
}

//Math
int missingNumber2(vector<int>& nums) {
    int n = nums.size();
    int sum = 0;
    for(int i = 0; i < nums.size(); i++)
        sum += nums[i];
    return (1+n)*n/2 - sum;
}


//Interesting
int missingNumber3(vector<int>& nums) {
    int sum = 0;
    for(int i = 0; i < nums.size(); i++)
        sum += (nums[i]-i);
    return nums.size() - sum;
}
